.travis.yml

```diff
@@ -65,23 +63,38 @@ before_script:
     - if [ "$FLAVOR" == "cs" ]; then travis_retry nuget restore cs/cs.sln; fi
 
 script:
-    - mkdir build && cd build
+    - mkdir build
+
+    - if [[ "$FLAVOR" == cpp-* ]]; then
+        docker run -v $HOME/.ccache:/root/.ccache -v $HOME/.stack:/root/.stack -v `pwd`:/root/bond -i bond-xenial sh -c "
+            mkdir -p $HOME;
+            ln -s /root/.ccache $HOME/.ccache;
+            ln -s /root/.stack $HOME/.stack;
+            export BOOST_ROOT=/opt/boosts/boost_`echo $BOOST | tr . _`;
+            export CXX='ccache clang++ -Qunused-arguments --system-header-prefix=boost/';
+            export CC='ccache clang -Qunused-arguments --system-header-prefix=boost/';
+            cd /root/bond/build;
+            cmake $CPP_CMAKE_ARGS ..;
+            make --jobs 2 check;
+            ";
+      else
+        cd build;
```

Comment:
It feels a little cleaner to have each of the cs and has flavors do this cd themselves. That makes each flavor independent of the others.

Reply:
Updated.